apps/total-typescript/src/lib/bookmarks.ts

```diff
@@ -0,0 +1,158 @@
+import {z} from 'zod'
+import {prisma} from '@skillrecordings/database'
+
+export const BookmarkInputSchema = z.object({
+  userId: z.string(),
+  resourceId: z.string(),
+  type: z.string(),
+  fields: z.record(z.string().nullable()).optional(),
+})
+
+export const ReturnedBookmarkSchema = z.object({
+  id: z.string(),
+  resourceId: z.string(),
+  type: z.string(),
+  fields: z.record(z.string().nullable()).optional(),
+})
+
+export const ReturnedBookmarkForBookSchema = z.object({
+  id: z.string(),
+  resourceId: z.string(),
+  type: z.string(),
+  fields: z.object({
+    chapterSlug: z.string(),
+    chapterTitle: z.string(),
+    resourceTitle: z.string(),
+  }),
+})
```

Comment:
when we know we are working with a Books bookmark we can get more specific on the type being returned